sycl/tools/syclbin-dump/syclbin-dump.cpp

```diff
@@ -0,0 +1,138 @@
+//==----------- syclbin-dump.cpp -------------------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// The "syclbin-dump" utility lists the contents of a SYCLBIN file in a
+// human-readable format.
+//
+
+#include "llvm/Object/OffloadBinary.h"
+#include "llvm/Object/SYCLBIN.h"
+#include "llvm/Support/CommandLine.h"
+
+#include <fstream>
+#include <iostream>
+#include <iterator>
+#include <string>
+#include <utility>
+
+using namespace llvm;
+
+std::string_view StateToString(llvm::object::SYCLBIN::BundleState State) {
+  switch (State) {
+  case llvm::object::SYCLBIN::BundleState::Input:
+    return "input";
+  case llvm::object::SYCLBIN::BundleState::Object:
+    return "object";
+  case llvm::object::SYCLBIN::BundleState::Executable:
+    return "executable";
+  default:
+    return "UNKNOWN";
+  }
+}
+
+std::string_view IRTypeToString(llvm::object::SYCLBIN::IRType IRType) {
+  switch (IRType) {
+  case llvm::object::SYCLBIN::IRType::SPIRV:
+    return "SPIR-V";
+  case llvm::object::SYCLBIN::IRType::PTX:
+    return "PTX";
+  case llvm::object::SYCLBIN::IRType::AMDGCN:
+    return "AMDGCN";
+  default:
+    return "UNKNOWN";
+  }
+}
+
+int main(int argc, char **argv, char *env[]) {
+  std::ignore = env;
```

Comment:
```suggestion
int main(int argc, char **argv) {
```

?

Reply:
I kept it in case it would become useful, but I don't feel strongly one way or the other.